tests/render/test_text.py

```diff
@@ -441,35 +441,6 @@ def test_render_text_encoding(
     assert "\n".join(expected_output).strip() == captured.out.strip()
 
 
-def test_render_text_list_all_and_packages_options_used(
-    capsys: pytest.CaptureFixture[str],
-    mock_pkgs: Callable[[MockGraph], Iterator[Mock]],
-) -> None:
```

Comment:
Let's not delete this as this has to do with `pipdeptree --list --all` being used together.

Why was it failing in the first place?

Reply:
Sorry, I can't find the origin comment since there too many of them. I removed it because it has the same problem with `test_package_dag_from_pkgs`, which you commented that it should be removed.

Reply:
To be sure, I wanted to see if we can actually assume that `Distribution.requires()` will honor the raw project name. Turns out, it doesn't (though I can now finally see why). For example, pre-commit depends on `PyYAML`, but in its [config file](https://github.com/pre-commit/pre-commit/blob/7b4667e9e6e05e31707c404c95115b151745866c/setup.cfg#L25) it has it as `pyyaml`, so we get the following:
```python
>>> from importlib.metadata import distribution
>>> d = distribution('pre-commit')
>>> d.requires
['cfgv (>=2.0.0)', 'identify (>=1.0.0)', 'nodeenv (>=0.11.1)', 'pyyaml (>=5.1)', 'virtualenv (>=20.10.0)']
>>> 
```

Just to better see where we are getting it from, its METADATA file has it as the following:
```text
Requires-Dist: pyyaml (>=5.1)
```

So now, what we need to do is do the same thing I did in the past and that is use the installed name if it exists, otherwise fallback to the name given in the METADATA file. I'll make a code suggestion in the proper place. Once  you implement this, this test should pass.

Reply:
Yes, what `Distribution.requires()`  has done is parsing the metadata file of distributions, so it only returns the name in that.
when you said we should remove this, I thought we now  only need to print the name in metadata.
I will bring this part back.